### PR TITLE
optical rotation code

### PR DESCRIPTION
**Pull request by colemanliyah**
Extracts optical rotation values from pubchem list of strings where each string is information on one molecule. Currently, the code works to extract the optical rotation values from the raw data file given. But the code can be slightly altered to take in a different data file. 

All that's included in this PR is given in the manifest.toml file.

### CODE REVIEW

File: optical_rotation_scrapping/manifest.toml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Manifest for optical_rotation_scrapping
+[source]
+title = "Scrapping optical rotation values from molecules in PubChem"
+authors = "Richard R Gerkin, Liyah A Coleman"
```
Comment (rgerkin) [inline optical_rotation_scrapping/manifest.toml:4]: My middle initial is C

File: optical_rotation_scrapping/manifest.toml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# Manifest for optical_rotation_scrapping
+[source]
+title = "Scrapping optical rotation values from molecules in PubChem"
```
Comment (rgerkin) [inline optical_rotation_scrapping/manifest.toml:3]: Make the title describe the data not the process

File: optical_rotation_scrapping/manifest.toml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+authors = "Richard R Gerkin, Liyah A Coleman"
+
+[raw]
+Values were extracted from PubChem
```
Comment (rgerkin) [inline optical_rotation_scrapping/manifest.toml:7]: This should have a file name in quotes, then =, then a description in quotes.  Add a bit more in the description (i.e. the heading from which they were extracted).

File: optical_rotation_scrapping/manifest.toml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+[processed]
+"opticalrotation.py" = "code for scrapping"
+"opticalroation.ipynb" = "notebook for running code"
```
Comment (rgerkin) [inline optical_rotation_scrapping/manifest.toml:11]: The file that runs the code to generated the processed data should always be "main.py" or "main.ipynb" for consistency.

The list of molecules should be molecules.py, the list of values should be in properties.py.  This is based on making shot-in-dark loading of these data successful without referring back to the details of the manifest.

**Review by rgerkin — COMMENTED** [review]
See the comments here and also add a molecules.csv (based on piping CIDs through pyrfume.odorants.get_cids and wrapping in a dataframe) and an attributes.csv which actually contain the OPR values (the column name can be "Optical Rotation Angle" or something like that).

File: optical_rotation_scrapping/manifest.toml [hunk 5]
```diff
@@ -1,12 +1,13 @@
...
 
 [processed]
-"opticalrotation.py" = "code for scrapping"
-"opticalroation.ipynb" = "notebook for running code"
-"opr_values.csv" = "dataframe of the scrpped opr values"
\ No newline at end of file
+"main.py" = "code for scrapping"
+"main.ipynb" = "notebook for running code"
+"molecules.csv" = "Molecules used"
+"attributes.csv" = "dataframe of the scrpped opr values"
```
Comment (rgerkin) [inline optical_rotation_scrapping/manifest.toml:13]: Fix typo and capitalize first word of each sentence.

**Review by rgerkin — CHANGES_REQUESTED** [review]
Please make these fixes.

File: optical_rotation_scrapping/attributes.csv [hunk 6]
```diff
@@ -0,0 +1,1459 @@
+CIDs,Optical Rotation Angle
```
Comment (rgerkin) [inline optical_rotation_scrapping/attributes.csv:1]: First column should be called "CID", not "CIDs"

File: optical_rotation_scrapping/manifest.toml [hunk 7]
```diff
@@ -0,0 +1,13 @@
+# Pubchem molecules with optical rotation values
+[source]
+title = "Scraping optical rotation values from molecules in PubChem"
+authors = "Richard C Gerkin, Liyah A Coleman"
```
Comment (rgerkin) [inline optical_rotation_scrapping/manifest.toml:4]: You should list yourself first.

File: optical_rotation_scrapping/molecules.csv [hunk 8]
```diff
@@ -0,0 +1,1459 @@
+,CID,MolecularWeight,IsomericSMILES,IUPACName,name
```
Comment (rgerkin) [inline optical_rotation_scrapping/molecules.csv]: The CID column itself should be the index, so you want to set it as the index before you export the csv file (this way it matches the attributes file).  So there shouldn't be a separate integer index (0, 1, 2, ...).

File: optical_rotation_scrapping/raw_data.csv [hunk 9]
```diff
@@ -0,0 +1,1459 @@
+CIDs,Properties,Properties,Properties,Properties,Properties
```
Comment (rgerkin) [inline optical_rotation_scrapping/raw_data.csv]: Change to "CID".  Also why are there 5 columns called "Properties"?  Are there up to 5 strings for each CID?  If so, change them to "Properties1", "Properties2", etc.

**Review by rgerkin — CHANGES_REQUESTED** [review]

File: optical_rotation_scrapping/attributes.csv [hunk 10]
```diff
@@ -1,4 +1,4 @@
-CIDs,Optical Rotation Angle
+,Optical Rotation Angle
```
Comment (rgerkin) [inline optical_rotation_scrapping/attributes.csv:1]: Index column name should be "CID", not blank.

File: optical_rotation_scrapping/attributes.csv [hunk 11]
```diff
@@ -13,7 +13,7 @@ CIDs,Optical Rotation Angle
 3032576,+230
```
Comment (rgerkin) [inline optical_rotation_scrapping/attributes.csv:13]: Why are there so many CIDs associated with the same rotation angle?  e.g. lines 13-19 below, or lines 93-102, etc.

**Review by rgerkin — CHANGES_REQUESTED** [review]
One small change and also please help me understand why there are so many duplicate angles (see comment).

File: optical_rotation_scrapping/attributes.csv (hunk 11 above)
Comment (colemanliyah) [reply optical_rotation_scrapping/attributes.csv:13]: For these molecules, their opr value just happened to be the same. For instance, in the case of the molecule in row 13 (https://pubchem.ncbi.nlm.nih.gov/compound/3032576#section=Optical-Rotation) and 14 (https://pubchem.ncbi.nlm.nih.gov/compound/3036746#section=Optical-Rotation) they had a few opr values listed but I take the one that was recorded at a temperature closest to room temperature. But the values for both seem to be the same.

Comment (rgerkin) [reply optical_rotation_scrapping/attributes.csv:13]: Oh, interesting.  It turns out it is the same molecule but with different PubChem IDs.  I think this isn't possible anymore (if they identified the same molecule they would just update the original entry) but maybe back in the day they weren't so good about catching this.

**Pull request merged**